warp10/src/main/sh/warp10.sh

```diff
@@ -525,133 +391,168 @@ start() {
 }
 
 stop() {
-
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
-
+  isWarp10User
   if isStarted; then
     echo "Stopping Warp 10..."
-    kill $(cat ${PID_FILE})
+    kill "$(cat "${PID_FILE}")"
     echo "Waiting for Warp 10 to stop..."
-    while $(kill -0 $(cat ${PID_FILE}) 2>/dev/null); do
+    while $(kill -0 $(cat "${PID_FILE}") 2>/dev/null); do
       sleep 2
     done
     echo "Warp 10 stopped..."
-    rm -f ${PID_FILE}
+    rm -f "${PID_FILE}"
   else
     echo "No instance of Warp 10 is currently running"
   fi
 }
 
 status() {
-
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
-
   if isStarted; then
-    ps -Ao pid,etime,args | grep "^\s*$(cat ${PID_FILE})\s"
+    ps -Ao pid,etime,args | grep "^\s*$(cat "${PID_FILE}")\s"
   else
     echo "No instance of Warp 10 is currently running"
   fi
-  
+
   checkRam
 }
 
 tokengen() {
   if [ "$#" -ne 2 ]; then
-    echo "Usage: $0 tokengen envelope.mc2"
-    exit 1
+    die "Usage: $0 tokengen envelope.mc2"
   fi
-  getConfigFiles
-  ${JAVACMD} -cp ${WARP10_CP} -Dlog4j.configuration=file:${LOG4J_CONF} -Dfile.encoding=UTF-8 io.warp10.TokenGen ${CONFIG_FILES} $2
+  ${JAVACMD} -cp "${WARP10_JAR}" -Dlog4j.configuration=file:"${LOG4J_CONF}" -Dfile.encoding=UTF-8 io.warp10.TokenGen ${CONFIG_FILES} "$2" 2>/dev/null
```

Comment:
This should go beyond the simple JAR as there may be some extensions which are loaded by the configuration. This is why the classpath was used.

Reply:
done